overlord/servicestate/quota_handlers.go

```diff
@@ -156,21 +157,32 @@ func (m *ServiceManager) doQuotaControl(t *state.Task, _ *tomb.Tomb) error {
 		}
 	}
 
+	log.Println("[QUOTA]", "quota-control", len(servicesAffected), refreshProfiles)
 	if len(servicesAffected) > 0 {
-		prevTask := t
+		var prevTaskSet *state.TaskSet
+		chg := t.Change()
+		queueTasks := func(ts *state.TaskSet) {
+			if prevTaskSet != nil {
+				prevTaskSet.WaitAll(ts)
+			} else {
+				ts.WaitFor(t)
+			}
+			chg.AddAll(ts)
+			prevTaskSet = ts
```

Comment:
this reads a bit nicer but still doesn't solve the problem if we had tasks waiting on quota-control itself, that's the HaltTasks handling part of InjectTasks

Reply:
I'm also not sure I understand why InjectTasks can't be used for this?

Reply:
what I would have expect is to create one TaskSet with possibly the setup-profiles and restarts and then injecting in one go with InjectTasks

Reply:
Fixed! I have revised this to now creating one task set and injecting them with InjectTasks.